### ISSUE TITLE
PSR12.Files.FileHeader misidentifies file header in mixed content file

### ISSUE THREAD

**Issue by okainov** [PHPCSStandards/PHP_CodeSniffer#1330]
## Describe the bug

I'm getting ` 2 | ERROR | The file header must be the first content in the file (PSR12.Files.FileHeader.HeaderPosition)` when `<?php` is in the middle of a mixed file when followed by docstring-like comment.

### Code sample
```php
<div></div>
<?php

/*** Comment - this is important, only comment in such format triggers the issue ***/

$b = 3;

```

### Custom ruleset
N/A

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs -s --standard=PSR12 test.php`
3. See error message displayed
```text
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------
 2 | ERROR | The file header must be the first content in the file (PSR12.Files.FileHeader.HeaderPosition)
```

## Expected behavior

No error expected in this case

## Versions (please complete the following information)

|                                          |                                                                                                                |
|-----------------------------|---------------------------------------------------------------------------- |
| Operating System            | Windows, Docker                                                          |
| PHP version                      | 7.4                                                                                         |
| PHP_CodeSniffer version | PHP_CodeSniffer version 3.13.2 (stable) , 4.0.1 (stable)                                                 |
| Standard                          |PSR12                                                         |
| Install type                       | Composer in docker container |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `4.x` branch of PHP_CodeSniffer.

See also https://github.com/squizlabs/PHP_CodeSniffer/issues/2732

**Comment by jrfnl** [comment]
✅ Confirmed as reproducable.
Solution to be determined.

Similar issues:
* https://github.com/squizlabs/PHP_CodeSniffer/issues/2911
* https://github.com/squizlabs/PHP_CodeSniffer/issues/2848
* https://github.com/squizlabs/PHP_CodeSniffer/issues/2732

Other related issues:
* https://github.com/squizlabs/PHP_CodeSniffer/issues/3596
* https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/21 ( https://github.com/squizlabs/PHP_CodeSniffer/issues/2997 )


I believe the sniff may need a careful review to verify it correctly complies with the [following rule from PSR12](https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements):

> When a file contains a mix of HTML and PHP, any of the above sections may still be used. If so, they MUST be present at the top of the file, even if the remainder of the code consists of a closing PHP tag and then a mixture of HTML and PHP.
